src/OpenSage.Mods.Generals/Gui/GeneralsControlBar.cs

```diff
@@ -118,7 +118,7 @@ public void Update(Player player)
 
             _moneyDisplay.Text = $"$ {player.Money}";
 
-            if (player.SelectedUnits.Count > 0)
+            if (player.SelectedUnits.Count > 0 && player.SelectedUnits.First().Owner == player)
```

Comment:
I think that is already part of a different PR of you?

Reply:
You're right.